spec/coc_spec.rb

```diff
@@ -0,0 +1,230 @@
+          state:      :consumption,
+          ext_id:     "dhis2acta.consumptioncoc",
+          name:       "inlined-dhis2acta.consumptioncoc",
+          origin:     "dataValueSets",
+          expression: '#{dhis2acta.consumptioncoc}'
+        ),
+        Orbf::RulesEngine::ActivityState.new_indicator(
+          state:      :stock_end,
+          ext_id:     "dhis2acta.stockendcoc",
+          name:       "inlined-dhis2acta.stockendcoc",
+          origin:     "dataValueSets",
+          expression: '#{dhis2acta.stockendcoc}'
+        )
+      ]
+    )
+  end
+
+  let(:package_b) do
+    Orbf::RulesEngine::Package.new(
+      code:                        :quantity_b,
+      kind:                        :single,
+      frequency:                   :quarterly,
+      activities:                  [activity_b],
+      main_org_unit_group_ext_ids: ["contracgroup1"],
+      rules:                       [
+        Orbf::RulesEngine::Rule.new(
+          kind:     :activity,
+          formulas: [
+            Orbf::RulesEngine::Formula.new(
+              "score", "round(safe_div(consumption, stock_end) * 100,2)", "",
+              activity_mappings: {
+                "actb" => "dhis2actb.scorecoc"
+              }
+            )
+
+          ]
+        )
+      ]
+    )
+  end
+
+  let(:payment_rule) do
+    Orbf::RulesEngine::PaymentRule.new(
+      code:      "pbf_payment",
+      frequency: :quarterly,
+      packages:  [package_a, package_b],
+      rule:      Orbf::RulesEngine::Rule.new(
+        kind:     "payment",
+        formulas: [
+          Orbf::RulesEngine::Formula.new(
+            "nothing",
+            "1",
+            "",
+            single_mapping: "payment_de"
+          )
+        ]
+      )
+    )
+  end
+
+  let(:project) do
+    Orbf::RulesEngine::Project.new(
+      packages:      [
+        package_a,
+        package_b
+
+      ],
+      payment_rules: [
+        payment_rule
+      ],
+      dhis2_params:  {
+        url:      "https://admin:district@play.dhis2.org/2.28",
+        user:     "admin",
+        password: "district"
+      }
+    )
+  end
+
+  describe "when data" do
+    let(:fetch_and_solve) do
+      Orbf::RulesEngine::FetchAndSolve.new(
+        project,
+        "1",
+        "2018Q1",
+        pyramid:     pyramid,
+        mock_values: [
+          {
+            "orgUnit"             => "1",
+            "period"              => "2018Q1",
+            "dataElement"         => "dhis2acta",
+            "categoryOptionCombo" => "stockstartcoc",
+            "value"               => "8"
+          },
+          {
+            "orgUnit"             => "1",
+            "period"              => "2018Q1",
+            "dataElement"         => "dhis2acta",
+            "categoryOptionCombo" => "stockendcoc",
+            "value"               => "6"
+          }
+        ]
+      )
+    end
+
+    it "exports nil values to dhis2 when no dhis2 values" do
```

Comment:
(I think that lines needs better words)

But this might also need a comment for future me, something like:

``` ruby
# Package A will read from `dhis2acta` got get the `stock_start`
# (`stockstartcoc`) and `stock_end` (`stockendcoc`), it will then
# output a `comsumption` to `"dhis2acta.consumptioncoc"`.
#
# Package B will read from `dhis2acta` to get the `stock_end`
# (`stockendcoc`) and from the newly created
# `dhis2acta.consumptioncoc`
#
# So putting this all together, if `start_stock` is 8 and `stock_end`
# is 6, the `consumption` will be 2 and that will be stored in
# `dhis2acta.consumptioncoc`
```